dask/_task_spec.py

```diff
@@ -692,6 +692,32 @@ def is_coro(self):
                 self._is_coro = False
         return self._is_coro
 
+    @staticmethod
+    def fuse(*tasks: Task, key: KeyType | None = None) -> Task:
+        leafs = set()
+        all_keys = set()
+        all_deps: set[KeyType] = set()
+        for t in tasks:
+            if t.key not in all_deps:
+                leafs.add(t.key)
+            all_deps.update(t.dependencies)
+            all_keys.add(t.key)
+            leafs -= t.dependencies
+        external_deps = all_deps - set(all_keys)
```

Comment:
nit: Isn't `all_keys` already a `set`?
```suggestion
        external_deps = all_deps - all_keys
        
```

Reply:
yes, I think I had a version where i was using a dict